ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DasSamplerBasic.java

```diff
@@ -0,0 +1,112 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.statetransition.datacolumns;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.units.bigints.UInt256;
+import tech.pegasys.teku.ethereum.events.SlotEventsChannel;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.datastructures.blobs.versions.eip7594.DataColumnSidecar;
+import tech.pegasys.teku.spec.datastructures.networking.libp2p.rpc.DataColumnIdentifier;
+import tech.pegasys.teku.spec.datastructures.state.Checkpoint;
+import tech.pegasys.teku.spec.logic.versions.eip7594.helpers.MiscHelpersEip7594;
+import tech.pegasys.teku.storage.api.FinalizedCheckpointChannel;
+
+public class DasSamplerBasic
+    implements DataAvailabilitySampler, FinalizedCheckpointChannel, SlotEventsChannel {
+  private static final Logger LOG = LogManager.getLogger("das-nyota");
+
+  private final UInt256 nodeId;
+  private final int totalCustodySubnetCount;
+  private final DataColumnSidecarCustody custody;
+
+  private final Spec spec;
+  private final DataColumnSidecarDB db;
+
+  public DasSamplerBasic(
+      final Spec spec,
+      final DataColumnSidecarDB db,
+      final DataColumnSidecarCustody custody,
+      final UInt256 nodeId,
+      final int totalCustodySubnetCount) {
+    checkNotNull(spec);
+    checkNotNull(db);
+    this.spec = spec;
+    this.db = db;
+    this.custody = custody;
+    this.nodeId = nodeId;
+    this.totalCustodySubnetCount = totalCustodySubnetCount;
+  }
+
+  @Override
+  public SafeFuture<List<DataColumnSidecar>> checkDataAvailability(
+      UInt64 slot, Bytes32 blockRoot, Bytes32 parentRoot) {
+    LOG.info("Requested checkDataAvailability for {}({})", blockRoot, slot);
+    final Optional<MiscHelpersEip7594> maybeMiscHelpers =
+        spec.atSlot(slot).miscHelpers().toVersionEip7594();
+    final Optional<List<UInt64>> columnIndexes =
+        maybeMiscHelpers.map(
+            miscHelpersEip7594 ->
+                miscHelpersEip7594.computeCustodyColumnIndexes(nodeId, totalCustodySubnetCount));
+    if (columnIndexes.isEmpty()) {
+      return SafeFuture.completedFuture(List.of());
+    }
+
+    return SafeFuture.collectAll(
+            columnIndexes.get().stream()
+                .map(
+                    index ->
+                        custody.getCustodyDataColumnSidecar(
```

Comment:
Assuming that all columns in our DB are inherently valid, may be it is cheaper to just query for available column indexes?

Reply:
https://github.com/Nashatyrev/teku/issues/123